src/main/java/core/basesyntax/dao/StorageDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.storage.Storage;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+    private Storage storage;
+
+    public StorageDaoImpl() {
+        storage = new Storage();
+    }
+
+    @Override
+    public void save(String fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("fruit cannot be null");
+        }
+        storage.getFruitStorage().put(fruit, quantity);
```

Comment:
When saving a fruit, consider checking if the fruit already exists in the storage and updating the quantity accordingly, rather than overwriting it.